### ISSUE TITLE
CancellablePromise.all(iterable) throws an error if a item of the iterable parameter is a boolean

### ISSUE THREAD

**Issue by fernandosouza** [metal/metal-promise#7]
According to the specification, the method all() is able to accept a boolean as one item of the iterable parameter. Metal-promise throws an error. 

`Uncaught TypeError: promise.then is not a function`

https://www.promisejs.org/api/

**Comment by eduardolundgren** [comment]
What would expect in this situation? If you pass a boolean to the `.all` it must fail.

**Comment by fernandosouza** [comment]
if I am not wrong, it should behave like a fulfilled promise once the snippet in the documentation calls the `then` method.

```
var promise = Promise.resolve(3);
Promise.all([true, promise]).then(values => {
  console.log(values); // [true, 3]
});
```

**Comment by fernandosouza** [comment]
Apparently, if you pass a value that it is not a promise, it creates one promise that is resolved to the given value.

<img width="832" alt="screen shot 2017-01-30 at 15 06 12" src="https://cloud.githubusercontent.com/assets/502575/22435158/b219004e-e6fd-11e6-981a-3e104bdb8b26.png">

**Comment by eduardolundgren** [comment]
That make sense. We need to check in Google Closure library if they have fixed it and bring the code back since we use their implementation.

**Comment by jbalsas** [comment]
I think this might be covered in the [`testAllWithNonThenable`](https://github.com/google/closure-library/blob/master/closure/goog/promise/promise_test.js#L1093-L1104), if so, it should've been fixed at some point.

**Comment by fernandosouza** [comment]
You are right, @jbalsas. This test case covers this case.

**Comment by jbalsas** [comment]
This seems to be the initial commit that introduced the feature: https://github.com/google/closure-library/commit/2c41c53cd54dc87773ad679e69e542851e19f68e

**Comment by jbalsas** [comment]
@mairatma, @eduardolundgren, I'm not sure if we documented the process to update this, since code is slightly different than the original (es6, bundled and no reference to `goog.require`...).

**Comment by mairatma** [comment]
Right now it's actually a very manual process of identifying what changed and doing the changes directly in the bundle, unfortunately. I couldn't find a way to automate the process that wouldn't add a lot more code than necessary to the bundle. For example, using the closure compiler we'd definitely receive a bundle that works, but it would be huge. Unfortunately I could never think of a better way to do this.

**Comment by mairatma** [comment]
Oh actually I was talking of metal-soy-bundle here. For this promise module it should still be manual, but a lot simpler, since it's more isolated, but I think we've never had to update it before.

**Comment by eduardolundgren** [comment]
This one was a bit easier since we got the implementation from Google and replaced the `goog.` utilities. I can show to you @fernandosouza how I did it.

**Comment by fernandosouza** [comment]
If you have any advice I would like to hear, @eduardolundgren. But it seems we have only figure out the changes since we've last brought the code. Right? I've already taken a note to remind me to do that.

**Comment by eduardolundgren** [comment]
@fernandosouza I would bring it all again, no need for finding what has changed inside.

**Comment by jbalsas** [comment]
Keep in mind some additional changes that were applied overtime (not sure all of them are still necessary):
- [Uses named imports instead of "core" object](https://github.com/metal/metal-promise/commit/b92e4f4b3c8ccbe3a52cae56125ba9ac1df00b00)
- [Remove define annotation since it will break for amd builds](https://github.com/metal/metal-promise/commit/ca7a6903aadff2cd038a9c3224faeccf42e2c4b8) and [Adds back documentation text](https://github.com/metal/metal-promise/commit/7c3c06a59d884ccb1dc1ef0491986be2f4a470ac)
- [Removing duplicated code (metal.async)](https://github.com/metal/metal-promise/commit/03c9edfb34dc7403e62ceca2a9a4f1456405309b)
- [Replaces instanceof calls to CancellationError with an object property](https://github.com/metal/metal-promise/commit/0e911032d643fda75cd011415fb972527e9998e1)
- ... (and others) ...

**Comment by jbalsas** [comment]
This has been fixed and merged to `develop`, to be released in `3.0.3`
